### PR TITLE
TST: Don't cast in test_cython_agg_nullable_int

### PR DESCRIPTION
**Pull request by mroeschke**
xref https://github.com/pandas-dev/pandas/pull/54129/files#r1272774080

Nice find @rhshadrach!

### CODE REVIEW

File: pandas/tests/groupby/aggregate/test_cython.py [hunk 1]
```diff
@@ -336,15 +336,9 @@ def test_cython_agg_nullable_int(op_name):
         }
     )
     result = getattr(df.groupby("A")["B"], op_name)()
-    df2 = df.assign(B=df["B"].astype("float64"))
+    df2 = df.assign(B=df["B"])
```
Comment (rhshadrach) [inline pandas/tests/groupby/aggregate/test_cython.py:339]: I think we want to still use NumPy dtypes here, then cast expected to the expected nullable dtype.

Comment (mroeschke) [reply pandas/tests/groupby/aggregate/test_cython.py]: Ah okay. I put back this cast and used `convert_dtypes` to cast expected to the nullable dtype

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @mroeschke

**Pull request merged**